packages/ia-pic-uploader/src/ia-pic-uploader.ts

```diff
@@ -0,0 +1,834 @@
+      false
+    );
+
+    // execute when uer drop image with ia-pic-uploader type version
+    this.selfSubmitEle?.addEventListener(
+      'drop',
+      this.handleDropImage.bind(this),
+      false
+    );
+
+    // execute when submit to save picture
+    this.saveFile?.addEventListener(
+      'submit',
+      this.handleSaveFile.bind(this),
+      false
+    );
+
+    // execute when user change picture
+    this.fileSelector?.addEventListener('change', () => {
+      const { files } = this.fileSelector!;
+      this.handleSelectedFiles(files);
+    });
+
+    document?.addEventListener('saveProfileAvatar', (e: Event) => {
+      if (this.fileSelector?.files.length) {
+        this.handleSaveFile(e);
+      }
+    });
+  }
+
+  preventDefault(e: Event) {
+    e.preventDefault();
+    e.stopPropagation();
+  }
+
+  /**
+   * execute when user drop image on input[type=file]
+   * @param event
+   */
+  handleDropImage(event: DragEvent) {
+    this.preventDefault(event);
+    this.selfSubmitEle?.classList.remove('drag-over');
+
+    if (!this.showLoadingIndicator) {
+      const files = event?.dataTransfer?.files;
+      if (files?.length) {
+        this.handleSelectedFiles(files);
+      }
+    }
+  }
+
+  /**
+   * display selected picture for preview
+   *
+   * @param {File} image
+   * @memberof IAPicUploader
+   */
+  previewImage(image: File) {
+    this.showDropper = true;
+    let img: HTMLImageElement;
+
+    if (this.type === 'full') {
+      img = document.createElement('img');
+    } else {
+      img = this.dropRegion?.querySelector('img') as HTMLImageElement;
+    }
+
+    img.alt = 'profile picture';
+
+    const reader = new FileReader();
+    reader.onload = e => {
+      img.src = e.target?.result as string;
+    };
+
+    if (this.type === 'full') {
+      const preview = this.selfSubmitEle?.querySelector('.full-preview');
+      preview?.appendChild(img);
+    }
+
+    // read the image...
+    reader.readAsDataURL(image);
+
+    this.dispatchEvent(new Event('fileChanged'));
+  }
+
+  /**
+   * validate the selected file extension and size
+   *
+   * @param {File} image
+   * @return {Boolean}
+   * @memberof IAPicUploader
+   */
+  validateImage(image: File): boolean {
+    this.fileValidationError = '';
+
+    // check the type
+    const validTypes = ['image/jpeg', 'image/png', 'image/gif'];
+
+    // check the size more than 5MB
+    const maxSizeInBytes = 5 * 1024 * 1024;
+
+    if (validTypes.indexOf(image.type) === -1) {
+      this.fileValidationError = 'file must be  format of JPEG or PNG or GIF.';
+      return false;
+    }
+
+    if (image.size > maxSizeInBytes) {
+      this.fileValidationError = 'file size must be less than 5MB.';
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * validate and preview selected image
+   *
+   * @param {(FilesModel | any)} files
+   * @memberof IAPicUploader
+   */
+  async handleSelectedFiles(files: FilesModel | any) {
```

Comment:
Q: `FilesModel | any` is the same as `any`. Can we be more specific about the type expected here?